src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -152,5 +185,77 @@ private static Summary[] RunWithExceptionHandling(Func<Summary[]> run)
+                throw new InvalidBenchmarkDeclarationException("Null not allowed.");
+
+            var nonBenchmarkTypes = types.Where(t => !t.ContainsRunnableBenchmarks()).ToArray();
+            if (!nonBenchmarkTypes.IsEmpty())
+            {
+                var invalidNames = string.Join("\n", nonBenchmarkTypes.Select(type => $"  {type.FullName}"));
+                throw new InvalidBenchmarkDeclarationException($"Invalid Types:\n{invalidNames}\nOnly public, non-generic (closed generic types with public parameterless ctors are supported), non-abstract, non-sealed, non-static types with public instance [Benchmark] method(s) are supported.");
+            }
+        }
+
+        private static void Validate(BenchmarkRunInfo[] benchmarkRunInfos)
+        {
+            if (benchmarkRunInfos.IsEmpty())
+                throw new InvalidBenchmarkDeclarationException($"No BenchmarkRunInfos provided.");
+
+            if (benchmarkRunInfos.Any(v => v is null))
+                throw new InvalidBenchmarkDeclarationException($"Null not allowed.");
+
+            foreach (var benchmarkRunInfo in benchmarkRunInfos)
+            {
+                if (benchmarkRunInfo.Config is null ||
+                    benchmarkRunInfo.BenchmarksCases is null ||
+                    benchmarkRunInfo.BenchmarksCases.IsEmpty() ||
+                    benchmarkRunInfo.BenchmarksCases.Any(c => c is null))
+                    throw new InvalidBenchmarkDeclarationException("BenchmarkRunInfo do not support null values.");
+            }
+        }
+
+        private static void Validate(Type type)
+        {
+            if (!type.ContainsRunnableBenchmarks())
+                throw new InvalidBenchmarkDeclarationException($"Type {type} is invalid. Only public, non-generic (closed generic types with public parameterless ctors are supported), non-abstract, non-sealed, non-static types with public instance [Benchmark] method(s) are supported.");
+        }
+
+        private static void Validate(Type type, MethodInfo[] methods)
+        {
+            if (!type.ContainsRunnableBenchmarks())
+                throw new InvalidBenchmarkDeclarationException($"Type {type} is invalid. Only public, non-generic (closed generic types with public parameterless ctors are supported), non-abstract, non-sealed, non-static types with public instance [Benchmark] method(s) are supported.");
+
+            if (methods.IsEmpty())
+                throw new InvalidBenchmarkDeclarationException($"No methods provided for {type}.");
+
+            if (methods.Any(m => m is null))
+                throw new InvalidBenchmarkDeclarationException($"Null not allowed.");
+
+            var benchmarkMethods = type.GetRunnableBenchmarks();
+            var invalidMethods = methods.Except(benchmarkMethods).ToArray();
+            if (!invalidMethods.IsEmpty())
+            {
+                var invalidNames = string.Join("\n", invalidMethods.Select(m => $"  {m.ReflectedType?.FullName}.{m.Name}"));
```

Comment:
Maybe someone know, how to get "global" methods that return null for m.ReflectedType?

Module.GetMethods() returns empty array for BDN, BCL(string/object) modules.

MSDN [says](https://docs.microsoft.com/en-us/dotnet/api/System.Reflection.MemberInfo.ReflectedType?view=netstandard-2.0#:~:text=If%20the%20MemberInfo%20object%20is%20a%20global%20member%20(that%20is%2C%20if%20it%20was%20obtained%20from%20the%20Module.GetMethods%20method%2C%20which%20returns%20global%20methods%20on%20a%20module)%2C%20the%20returned%20DeclaringType%20will%20be%20null.):
>If the MemberInfo object is a global member (that is, if it was obtained from the [Module.GetMethods](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.module.getmethods?view=netstandard-2.0) method, which returns global methods on a module), the returned [DeclaringType](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.declaringtype?view=netstandard-2.0) will be null.

Reply:
To check if given method belongs to the provided type you should be able to use `method.DeclaringType`

Reply:
Found the answer when it returns null (this applies to `DeclaringType` as well):

> It can be null if the property is defined in a module. In C# you cannot define such methods and properties without reflection (see PropertyBuilder). However, if you reference a VB.NET assembly, it can have such members.